### ISSUE TITLE
Is priority (salience) working properly?

### ISSUE THREAD

**Issue by macroach** [jruizgit/rules#94]
Hi @jruizgit, I've been playing with rule priority levels, and in the following example is not behaving the way I would expect. The conditions for both `rule1` and `rule2` are met after the initial fact is asserted, and so I would expect the one with the lower priority value (`rule2`) to fire, but `rule1` is triggered instead. Assuming this isn't a bug, would you mind explaining what is happening so that I can better understand the logic around rule priority/execution?

```
from durable.lang import *

with ruleset('ex3'):

    @when_all(
        pri(2),
        c.fact1 << m.toggle == True,
        none(+m.cancel)
    )
    def rule1(c):
        c.assert_fact({"cancel":True})
        print("rule1 fired")

    @when_all(
        pri(1),
        none(+m.cancel)
    )
    def rule2(c):
        c.assert_fact({"cancel":True})
        print("rule2 fired")


    @when_start
    def start(host):
        x = host.assert_fact("ex3",{"toggle":True})

run_all()
```
Output:
```
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
rule1 fired
```

Many thanks,
Mac

**Comment by jruizgit** [comment]
Hi @macroach, I took a quick look at the test case. I think salience is working OK. The problem is the `none` expression standalone is not triggering the rule. Let me debug and I will report back on my findings.

**Comment by jruizgit** [comment]
The current behavior of `none`: `none` requires data to reason about no data. The code below illustrates the point: With the first `assert_fact("ex3",{"cancel":True})` the engine blocks both rules. With the `retract_fact("ex3",{"cancel":True})` the engine unblocks both rules honoring the rule priority.

```python
from durable.lang import *

with ruleset('ex3'):

    @when_all(
        pri(2),
        c.fact1 << m.toggle == True,
        none(+m.cancel)
    )
    def rule1(c):
        c.assert_fact({"cancel":True})
        print("rule1 fired")

    @when_all(
        pri(1),
        none(+m.cancel)
    )
    def rule2(c):
        c.assert_fact({"cancel":True})
        print("rule2 fired")

    @when_start
    def start(host):
        host.assert_fact("ex3",{"cancel":True})
        host.assert_fact("ex3",{"toggle":True})
        host.retract_fact("ex3",{"cancel":True})

run_all()
```

**Comment by macroach** [comment]
Gotcha - thanks so much for looking into this and responding so quickly, @jruizgit ! 

Is this the intended behavior? (And if it is, is it documented somewhere? If so, I'll go ahead and mark the issue as closed.)

Thanks again!
Mac

**Comment by jruizgit** [comment]
Yes, it is the intended behavior of `none`. I have just added a note to the docs:

https://github.com/jruizgit/rules/blob/master/docs/py/reference.md#lack-of-information

Thanks,

**Comment by macroach** [comment]
Thanks!
